lib/Listener/BeforeTemplateRenderedListener.php

```diff
@@ -72,14 +73,18 @@ public function handle(Event $event): void {
 		// FIXME as long as photos app has photos as internal label and appname label
 		if ($response->getRenderAs() === TemplateResponse::RENDER_AS_USER ||
 			 $response->getRenderAs() === TemplateResponse::RENDER_AS_PUBLIC) {
+			$csp = new ContentSecurityPolicy();
 			$this->fixPhotosAppName();
+			$response->setContentSecurityPolicy($csp);
 		}
 
 		if (($response->getStatus() >= 400) && ($response->getStatus() < 600)) {
 			// render client error states with own layout => own #body-status id
+			$csp = new ContentSecurityPolicy();
 			$tmplparams = $response->getParams();
 			$tmplparams['bodyid'] = "body-status";
 			$response->setParams($tmplparams);
+			$response->setContentSecurityPolicy($csp);
```

Comment:
I doubt that an "empty" Content Security Policy has any effect if no CSP rules are added? Can you explain that?

Reply:
I have refactored the code, moving all relevant CSP Header setters into an event listener for Content Security Policy changes (which should happen every time a file is directly downloaded). Not sure the BeforeTemplateRenderedListener would have caught file download events, as that does not trigger the re-rendering of the template. Thank you to check and let me know what you think.